src/Message/ExtendedResponse.php

```diff
@@ -444,6 +456,59 @@ public function getTransaction()
         return $this->transaction;
     }
 
+    /**
+     * Returns refunds for the corresponding transaction that was fetched
+     *
+     * @return Refund[]|null
+     */
+    public function getRefunds()
+    {
+        if (empty($this->refunds)) {
+            $invoice = $this->getInvoice(Constants::REVERSAL_REFUND);
```

Comment:
Since we now have separate request and response classes for the Extended and Reporting APIs (yay!) we don't necessarily need `ExtendedResponse` and `ReportingResponse` to behave exactly the same way. What do you think about storing a single refund and a single chargeback on an instance of `ExtendedResponse` since the logic below implies that this array will only ever be an array of 1? It's not a huge deal but switching these methods to `getRefund` and `getChargeback` would more clearly communicate the fact that an Extended API response invoice only ever contains a single record, whereas the Reporting API is intended for retrieving multiple records.

Reply:
The naming for this might be deceptive. The refunds returned here will only be for a single transaction. However some transactions are refunded in parts (e.g. tax first and then the rest) The invoice will contain multiple refunded reversals for one transaction.

I took the same approach with Chargebacks although I couldn't find an example of a transaction with multiple chargebacks I figured it couldn't hurt to account for that case.

Reply:
I think I might be missing something here - when we have multiple refunded reversals, where/when do we get them from the invoice? This function appears to only get one refund from the invoice and I don't see anywhere else where we'd retrieve multiple?

Reply:
You're completely right! A single invoice can contain multiple <financial-transactions> and for some reason I assumed all the refunds would be included in there. But each separate refund will contain a separate <invoice> that needs to be extracted. I will need to change the `ExtendedResponse::invoice` property to support multiple invoices

Reply:
Aha, so what we really need support for is multiple invoices? Cool

Reply:
yeah. At least for chargebacks and refunds. Just noticed that github will hide anything in tags if you don't wrap it in backticks